api/src/main/java/org/pmiops/workbench/leonardo/LeonardoRetryHandler.java

```diff
@@ -36,12 +36,21 @@ protected int getResponseCode(Throwable lastException) {
     @Override
     protected void logNoRetry(Throwable t, int responseCode) {
       if (t instanceof ApiException) {
-        logger.log(
-            getLogLevel(responseCode),
-            String.format(
-                "Exception calling Leonardo API with response: %s",
-                ((ApiException) t).getResponseBody()),
-            t);
+        if (responseCode == 404) {
```

Comment:
One of the issues noted in the ticket is that we're incorrectly retrying GetRuntime multiple times - in fact we shouldn't be retrying 400-level errors except maybe in a few exceptional cases. Were you able to look into this?

Reply:
I must have overlooked that in the ticket description. This is definitely doable

Reply:
interesting: I don't think we're retrying anything but 500 series errors. I think what's happening is that we're calling `getRuntime` twice on the client, and then we're throwing twice per get. Can't figure out why. I don't think for instance commenting out `setThrowLastExceptionOnExhausted(true)` changes anything.